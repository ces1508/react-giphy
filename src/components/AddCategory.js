import React, { useState } from 'react'
import PropTypes from 'prop-types'

function AddCategory ({ setCategories }) {

  const [inputValue, setInputValue] = useState('')

  const handleInputChange = (e) => {
    const { value } = e.target
    setInputValue(value)
  }

  const handleSubmit = e => {
    e.preventDefault()
    if (!inputValue) return;

    setCategories(cats => [inputValue, ...cats])
    setInputValue('')
  }

  return (
    <form onSubmit={handleSubmit}>
      <input
        type='text'
        placeholder="Naruto, One Punch Man, Lol"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}

export default AddCategory
