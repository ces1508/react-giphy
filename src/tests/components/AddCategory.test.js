import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import AddCategory from '../../components/AddCategory'

describe('<AddCategory />', () => {

  const setCategories = jest.fn()
  let wrapper = shallow(<AddCategory setCategories={setCategories}/>); // use for editor helps

  beforeEach(() => {
    jest.clearAllMocks()
    wrapper = shallow(<AddCategory setCategories={setCategories}/>)
  })

  it('<AddCategory /> should be rendered', () =>{
    expect(wrapper).toMatchSnapshot()
  })


  it ('<AddCategory /> should be input value change when user put text', () => {
    const input = wrapper.find('input')
    const value = 'naruto'
    input.simulate('change', { target: { value } })

    const inputAfterChange = wrapper.find('input')
    expect(inputAfterChange.prop('value')).toBe(value)
  })

  it ('<AddCategory /> dont call setCategories function if inputValue is empty ', () => {
    const form = wrapper.find('form')

    form.simulate('submit', { preventDefault(){} })

    expect(setCategories).not.toHaveBeenCalled()

  })

  it ('<AddCategory /> setCategories will be called with parameters', () => {
    const input = wrapper.find('input')
    input.simulate('change', { target: { value: 'naruto' } })

    const form = wrapper.find('form')
    form.simulate('submit', { preventDefault(){} })

    const inputAfterSubmit = wrapper.find('input')
    expect(inputAfterSubmit.prop('value')).toBe('')
    expect(setCategories).toHaveBeenCalled()

  })

})
