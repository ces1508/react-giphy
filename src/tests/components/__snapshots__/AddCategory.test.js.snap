// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddCategory /> <AddCategory /> should be rendered 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Naruto, One Punch Man, Lol"
    type="text"
    value=""
  />
</form>
`;
